### PR TITLE
opencl-icd-loader: Add disable_openclon12 option to avoid Windows SDK

### PR DESCRIPTION
**Pull request by ItsBasi**
Specify library name and version:  **opencl-icd-loader/2021.04.29**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ItsBasi** [comment]
Khronos describes the option in the dependencies section. (https://github.com/KhronosGroup/OpenCL-ICD-Loader#dependencies)

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — COMMENTED** [review]
please, add required_conan_version, like on termcap: https://github.com/conan-io/conan-center-index/blob/ce0e51c1dac8efd736879181b3fd0ef37537b4a2/recipes/termcap/all/conanfile.py#L5

File: recipes/opencl-icd-loader/all/conanfile.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def requirements(self):
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
-        os.rename("OpenCL-ICD-Loader-" + self.version, self._source_subfolder)
+        tools.rename("OpenCL-ICD-Loader-" + self.version, self._source_subfolder)
```
[the remark this answers is not in the document]
Comment (ItsBasi) [reply recipes/opencl-icd-loader/all/conanfile.py]: Thanks for reminding me of strip_root.

File: recipes/opencl-icd-loader/all/conanfile.py [hunk 2]
```diff
@@ -48,6 +48,7 @@ def _configure_cmake(self):
             self._cmake.definitions["USE_DYNAMIC_VCXX_RUNTIME"] = str(self.settings.compiler.runtime).startswith("MD")
         self._cmake.definitions["OPENCL_ICD_LOADER_PIC"] = self.options.get_safe("fPIC", True)
         self._cmake.definitions["OPENCL_ICD_LOADER_BUILD_TESTING"] = False
+        self._cmake.definitions["OPENCL_ICD_LOADER_DISABLE_OPENCLON12"] = self.options.get_safe("disable_openclon12", False)
```
[the remark this answers is not in the document]
Comment (ItsBasi) [reply recipes/opencl-icd-loader/all/conanfile.py]: Commited

**Review by SSE4 — DISMISSED** [review]

File: recipes/opencl-icd-loader/all/conanfile.py [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
-    topics = ("conan", "opencl-icd-loader", "opencl", "khronos", "parallel", "icd-loader")
+    topics = ("opencl-icd-loader", "opencl", "khronos", "parallel", "icd-loader")
     homepage = "https://github.com/KhronosGroup/OpenCL-ICD-Loader"
     url = "https://github.com/conan-io/conan-center-index"
 
     settings = "os", "arch", "compiler", "build_type"
-    options = {"shared": [True, False], "fPIC": [True, False]}
-    default_options = {"shared": False, "fPIC": True}
+    options = {"shared": [True, False], "fPIC": [True, False], "disable_openclon12": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "disable_openclon12": False}
```
Comment (SpaceIm) [inline recipes/opencl-icd-loader/all/conanfile.py:16]: please, in `config_options()` remove this option if not windows, and inject it in `_configure_cmake()` if windows.

https://github.com/KhronosGroup/OpenCL-ICD-Loader/blob/4e65bd5db0a0a87637fddc081a70d537fc2a9e70/CMakeLists.txt#L94-L100

Comment (ItsBasi) [reply recipes/opencl-icd-loader/all/conanfile.py:16]: Commited

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**